crates/sui-core/src/authority.rs

```diff
@@ -905,6 +953,14 @@ impl AuthorityState {
         // commit_certificate finished, the tx is fully committed to the store.
         tx_guard.commit_tx();
 
+        // Notifies transaction manager about available input objects. This allows the transaction
+        // manager to schedule ready transactions.
+        // TODO: switch TransactionManager to use notify_read() on objects table.
```

Comment:
I'm not sure notify_read is really a better approach - we could potentially have a massive number of waiting readers.

In any case, this should be moved to before `tx_guard.commit_tx()` to ensure at-least-once execution.